### DISCUSSION TITLE
macos configure-tccdb-macos.sh failed: attempt to write a readonly database

### DISCUSSION THREAD

**Discussion by zoezhangmattr** [actions/virtual-environments#10174] [Image configuration]
i am trying to build macos runner locally using the runner images scripts
when it runs https://github.com/actions/runner-images/blob/0ad636b8243987e85a0148adcfd379b1661a0728/images/macos/templates/macOS-14.arm64.anka.pkr.hcl#L181
i got the error `Error: stepping, attempt to write a readonly database (8)`
did some exploration in the internet, all i found is tcc.db is readonly, if wanna modify it, need to disable SIP
i wonder how does github build it , i dont see anything related to SIP bypass configs.
any advice is appreciated!

**Comment by raghav155** [comment]
try to disable **System Integrity Protection** in your machine.

**Reply by zoezhangmattr** [reply]
thanks for your help, i just found its impossible to disable sip in amazon mac instance.
what would be impacted if this script is not executed.

**Reply by raghav155** [reply]
When the script tries to perform an action that requires Accessibility access, macOS may prompt the user to grant permission. This could lead failures while running the packer script (though I didn't try it).
